File: Ras Signature/Scripts/arhg_ras_connection.py
import os
from pathlib import Path
import matplotlib.pyplot as plt
import matplotlib.patches as mpatches
from matplotlib.patches import FancyBboxPatch
import matplotlib.patheffects as pe

fig, ax = plt.subplots(figsize=(14, 9))
ax.set_xlim(0, 14)
ax.set_ylim(0, 9)
ax.axis('off')
fig.patch.set_facecolor('white')

# ── Colors ──────────────────────────────────────────────
RED_FILL  = '#FDECEA'
RED_EDGE  = '#C0392B'
RED_HEAD  = '#FADBD8'
RED_TXT   = '#922B21'
RED_SUB   = '#A93226'

BLU_FILL  = '#EBF5FB'
BLU_EDGE  = '#2471A3'
BLU_HEAD  = '#D6EAF8'
BLU_TXT   = '#1A5276'
BLU_SUB   = '#21618C'

NAVY      = '#1B2A4A'
ARROW_CLR = '#7F8C8D'

# ── Helpers ─────────────────────────────────────────────
def box(x, y, w, h, fc, ec, radius=0.12, lw=0.9):
    p = FancyBboxPatch((x, y), w, h,
                       boxstyle=f"round,pad=0,rounding_size={radius}",
                       facecolor=fc, edgecolor=ec, linewidth=lw, zorder=2)
    ax.add_patch(p)

def txt(x, y, s, size, color='black', weight='normal', style='normal',
        ha='center', va='center'):
    ax.text(x, y, s, fontsize=size, color=color, fontweight=weight,
            fontstyle=style, ha=ha, va=va, linespacing=1.3, zorder=3)

def arrow(x, y_from, y_to, color=ARROW_CLR):
    ax.annotate('', xy=(x, y_to), xytext=(x, y_from),
                arrowprops=dict(arrowstyle='->', color=color,
                                lw=1.0, mutation_scale=11), zorder=3)

# ── Layout constants ─────────────────────────────────────
BH  = 0.70   # flow box height
BHH = 0.78   # header box height
GAP = 0.22   # vertical gap between boxes
INNER_W = 4.80

# Left column
LX, LW, LCX = 0.55, 6.10, 3.60
BXL = LX + (LW - INNER_W) / 2

# Right column
RX, RW, RCX = 7.35, 6.10, 10.40
BXR = RX + (RW - INNER_W) / 2

# ── Vertical positions (top of each element, y-axis up) ──
HEADER_TOP   = 8.33
BOX1_TOP     = 7.15
BOX2_TOP     = BOX1_TOP - BH - GAP    # 6.23
BOX3_TOP     = BOX2_TOP - BH - GAP    # 5.31
BOX4_TOP     = BOX3_TOP - BH - GAP    # 4.39
BOX5_TOP     = BOX4_TOP - BH - GAP    # 3.47

BOT_TOP      = BOX5_TOP - BH - GAP    # 2.55  (top of convergence box)
BOT_H        = 0.72
BOT_Y        = BOT_TOP - BOT_H        # 1.83

HYP_TOP      = BOT_Y - 0.17
HYP_H        = HYP_TOP - 0.06
HYP_Y        = HYP_TOP - HYP_H        # 0.06

# ════════════════════════════════════════════════════════
# TITLE
# ════════════════════════════════════════════════════════
txt(7, 8.65,
    'ARHG Mutations: Independent or Convergent Oncogenic Events?',
    11, '#1C1C1C', 'bold')

# ════════════════════════════════════════════════════════
# LEFT COLUMN — headers + 5 flow boxes
# ════════════════════════════════════════════════════════
# Header
box(LX, HEADER_TOP - BHH, LW, BHH, RED_HEAD, RED_EDGE, lw=1.1)
txt(LCX, HEADER_TOP - 0.26, '52% of ARHG-mutated patients',        9.5, RED_EDGE, 'bold')
txt(LCX, HEADER_TOP - 0.57, 'Co-mutated with NRAS / KRAS / KIT',   7.5, RED_SUB)

def left_box(top, title, subtitle):
    box(BXL, top - BH, INNER_W, BH, RED_FILL, RED_EDGE)
    txt(LCX, top - 0.24, title,    8.5, RED_TXT, 'bold')
    txt(LCX, top - 0.52, subtitle, 6.5, RED_SUB, style='italic')

L_BOXES = [
    (BOX1_TOP, 'ARHG mutation',
               'GEF gain-of-function  or  GAP loss-of-function'),
    (BOX2_TOP, 'NRAS / KRAS',
               'Direct RAS activation'),
    (BOX3_TOP, '\u2193 Rho GTPase activity',
               'RhoA / Rac1 / Cdc42'),
    (BOX4_TOP, '\u2193 RAS activity',
               'GTP-bound RAS'),
    (BOX5_TOP, 'PI3K / MAPK activation',
               'Convergent downstream signaling'),
]

for top, title, sub in L_BOXES:
    left_box(top, title, sub)

# Arrows between left boxes
for i in range(len(L_BOXES) - 1):
    arrow(LCX, L_BOXES[i][0] - BH, L_BOXES[i+1][0], RED_EDGE)

# ════════════════════════════════════════════════════════
# RIGHT COLUMN — headers + 3 flow boxes
# ════════════════════════════════════════════════════════
box(RX, HEADER_TOP - BHH, RW, BHH, BLU_HEAD, BLU_EDGE, lw=1.1)
txt(RCX, HEADER_TOP - 0.26, '48% of ARHG-mutated patients',                   9.5, BLU_EDGE, 'bold')
txt(RCX, HEADER_TOP - 0.57, 'No co-mutations in canonical RAS pathway genes',  7.5, BLU_SUB)

def right_box(top, title, subtitle):
    box(BXR, top - BH, INNER_W, BH, BLU_FILL, BLU_EDGE)
    txt(RCX, top - 0.24, title,    8.5, BLU_TXT, 'bold')
    txt(RCX, top - 0.52, subtitle, 6.5, BLU_SUB, style='italic')

R_BOXES = [
    (BOX1_TOP, 'ARHG mutation',
               'GEF gain-of-function  or  GAP loss-of-function'),
    (BOX2_TOP, '\u2193 Rho GTPase activity',
               'RhoA / Rac1 / Cdc42'),
    (BOX3_TOP, 'Cross talk \u2192 PI3K / MAPK',
               'via PI3K, RalGDS, NF-\u03baB'),
]

for top, title, sub in R_BOXES:
    right_box(top, title, sub)

# Small side note on Cross talk box
txt(BXR + INNER_W + 0.08, BOX3_TOP - BH/2,
    'Intrinsic\npathway\nsignaling',
    5.5, BLU_SUB, style='italic', ha='left')

# Arrows between right boxes
for i in range(len(R_BOXES) - 1):
    arrow(RCX, R_BOXES[i][0] - BH, R_BOXES[i+1][0], BLU_EDGE)

# ════════════════════════════════════════════════════════
# BOTTOM CONVERGENCE BOX
# ════════════════════════════════════════════════════════
BOT_X  = 1.1
BOT_W  = 11.8
BOT_CX = BOT_X + BOT_W / 2

box(BOT_X, BOT_Y, BOT_W, BOT_H, '#F4F6F7', '#95A5A6', lw=0.9)
txt(BOT_CX, BOT_Y + BOT_H*0.64,
    'Proliferation  \u00b7  Survival  \u00b7  Treatment resistance',
    9.5, '#2C3E50', 'bold')
txt(BOT_CX, BOT_Y + BOT_H*0.26,
    'Same downstream phenotype \u2014 independent of direct RAS mutation',
    6.5, '#5D6D7E', style='italic')

# Arrows to bottom box
arrow(LCX, L_BOXES[-1][0] - BH, BOT_Y + BOT_H, ARROW_CLR)
arrow(RCX, R_BOXES[-1][0] - BH, BOT_Y + BOT_H, ARROW_CLR)

# ════════════════════════════════════════════════════════
# HYPOTHESIS BAR
# ════════════════════════════════════════════════════════
box(0.45, HYP_Y, 13.1, HYP_H, NAVY, NAVY, radius=0.14, lw=0)

txt(7, HYP_Y + HYP_H * 0.67,
    'Hypothesis: ARHG mutations dysregulate Rho GTPase activity, '
    'feeding into RAS-adjacent signaling even without direct RAS mutations',
    7.5, 'white', 'bold', style='italic')
txt(7, HYP_Y + HYP_H * 0.28,
    'ARHG mutations may be independent oncogenic events \u2014 '
    'not simply passengers to NRAS or KRAS',
    6.0, '#BDC3C7')

# ════════════════════════════════════════════════════════
# SAVE
# ════════════════════════════════════════════════════════
plt.tight_layout(pad=0.2)
plt.savefig(
    str(Path(os.environ.get("ARHG_BASE_DIR", str(Path(__file__).parent.parent.parent))) / "Presentations/arhg_ras_connection.png"),
    dpi=180, bbox_inches='tight', facecolor='white'
)
print("Saved.")

File: Ras Signature/Scripts/arhg_ras_connection.R
library(ggplot2)
library(grid)

BASE_DIR <- Sys.getenv("ARHG_BASE_DIR", unset = normalizePath("."))

# Canvas dimensions
W <- 14
H <- 9

# Helper to draw a rounded-corner box
box <- function(xmin, xmax, ymin, ymax, fill, color, alpha = 1) {
  annotate("rect", xmin = xmin, xmax = xmax, ymin = ymin, ymax = ymax,
           fill = fill, color = color, linewidth = 0.6, alpha = alpha)
}

# Helper for centered text
lbl <- function(x, y, label, size, fontface = "plain", color = "black", hjust = 0.5, vjust = 0.5) {
  annotate("text", x = x, y = y, label = label, size = size,
           fontface = fontface, color = color, hjust = hjust, vjust = vjust,
           lineheight = 1.1)
}

# Arrow segment
arr <- function(x, xend, y, yend, color = "#555555") {
  annotate("segment", x = x, xend = xend, y = y, yend = yend,
           arrow = arrow(length = unit(0.12, "inches"), type = "closed"),
           color = color, linewidth = 0.6)
}

# ---- Colors ----
red_fill   <- "#FDECEA"
red_border <- "#C0392B"
red_head   <- "#FADBD8"

blue_fill   <- "#EBF5FB"
blue_border <- "#2471A3"
blue_head   <- "#D6EAF8"

navy_fill  <- "#1A2A4A"

# ---- Layout: x zones ----
# Left column center: x = 3.5, right column center: x = 10.5
# Figure runs x: 0.5 - 13.5

p <- ggplot() +
  xlim(0, W) + ylim(0, H) +
  theme_void() +
  theme(
    plot.background = element_rect(fill = "white", color = NA),
    plot.margin = margin(10, 10, 10, 10)
  )

# ===== TITLE =====
p <- p +
  lbl(7, 8.65, "ARHG Mutations: Independent or Convergent Oncogenic Events?",
      size = 6.5, fontface = "bold", color = "black")

# ===== LEFT COLUMN =====
lx1 <- 0.6; lx2 <- 6.6; lcx <- (lx1 + lx2) / 2

# Header box (red)
p <- p +
  box(lx1, lx2, 7.6, 8.35, red_head, red_border) +
  lbl(lcx, 8.1, "52% of ARHG-mutated patients", size = 5, fontface = "bold", color = red_border) +
  lbl(lcx, 7.82, "Co-mutated with NRAS / KRAS / KIT", size = 4, color = "#7B241C")

# Box A: ARHG mutation
p <- p +
  box(lx1 + 0.4, lx2 - 0.4, 6.6, 7.3, red_fill, red_border) +
  lbl(lcx, 7.05, "ARHG mutation", size = 4.5, fontface = "bold", color = red_border) +
  lbl(lcx, 6.78, "GEF gain-of-function  or  GAP loss-of-function", size = 3.3, color = "#7B241C")

# Arrow
p <- p + arr(lcx, lcx, 6.6, 6.35, red_border)

# Box B: NRAS / KRAS
p <- p +
  box(lx1 + 0.4, lx2 - 0.4, 5.6, 6.3, red_fill, red_border) +
  lbl(lcx, 6.03, "NRAS / KRAS", size = 4.5, fontface = "bold", color = red_border) +
  lbl(lcx, 5.76, "Direct RAS activation", size = 3.3, color = "#7B241C")

# Arrow
p <- p + arr(lcx, lcx, 5.6, 5.35, red_border)

# Box C: Rho GTPase
p <- p +
  box(lx1 + 0.4, lx2 - 0.4, 4.6, 5.3, red_fill, red_border) +
  lbl(lcx, 5.03, "\u2193 Rho GTPase activity", size = 4.5, fontface = "bold", color = red_border) +
  lbl(lcx, 4.76, "RhoA / Rac1 / Cdc42", size = 3.3, color = "#7B241C")

# Arrow
p <- p + arr(lcx, lcx, 4.6, 4.35, red_border)

# Box D: RAS activity
p <- p +
  box(lx1 + 0.4, lx2 - 0.4, 3.6, 4.3, red_fill, red_border) +
  lbl(lcx, 4.03, "\u2193 RAS activity", size = 4.5, fontface = "bold", color = red_border) +
  lbl(lcx, 3.76, "GTP-bound RAS", size = 3.3, color = "#7B241C")

# Arrow
p <- p + arr(lcx, lcx, 3.6, 3.35, red_border)

# Box E: PI3K / MAPK
p <- p +
  box(lx1 + 0.4, lx2 - 0.4, 2.6, 3.3, red_fill, red_border) +
  lbl(lcx, 3.03, "PI3K / MAPK activation", size = 4.5, fontface = "bold", color = red_border) +
  lbl(lcx, 2.76, "Convergent downstream signaling", size = 3.3, color = "#7B241C")

# ===== RIGHT COLUMN =====
rx1 <- 7.4; rx2 <- 13.4; rcx <- (rx1 + rx2) / 2

# Header box (blue)
p <- p +
  box(rx1, rx2, 7.6, 8.35, blue_head, blue_border) +
  lbl(rcx, 8.1, "48% of ARHG-mutated patients", size = 5, fontface = "bold", color = blue_border) +
  lbl(rcx, 7.82, "No co-mutations in canonical RAS pathway genes", size = 4, color = "#1A5276")

# Box A: ARHG mutation
p <- p +
  box(rx1 + 0.4, rx2 - 0.4, 6.6, 7.3, blue_fill, blue_border) +
  lbl(rcx, 7.05, "ARHG mutation", size = 4.5, fontface = "bold", color = blue_border) +
  lbl(rcx, 6.78, "GEF gain-of-function  or  GAP loss-of-function", size = 3.3, color = "#1A5276")

# Arrow
p <- p + arr(rcx, rcx, 6.6, 6.35, blue_border)

# Box B: Rho GTPase
p <- p +
  box(rx1 + 0.4, rx2 - 0.4, 5.6, 6.3, blue_fill, blue_border) +
  lbl(rcx, 6.03, "\u2193 Rho GTPase activity", size = 4.5, fontface = "bold", color = blue_border) +
  lbl(rcx, 5.76, "RhoA / Rac1 / Cdc42", size = 3.3, color = "#1A5276")

# Arrow
p <- p + arr(rcx, rcx, 5.6, 5.35, blue_border)

# Box C: Cross talk
p <- p +
  box(rx1 + 0.4, rx2 - 0.4, 4.6, 5.3, blue_fill, blue_border) +
  lbl(rcx - 0.3, 5.03, "Cross talk \u2192 PI3K / MAPK", size = 4.5, fontface = "bold", color = blue_border) +
  lbl(rcx - 0.3, 4.76, "Intrinsic pathway signaling", size = 3.3, color = "#1A5276") +
  # small note on right side
  lbl(rx2 - 0.5, 5.08, "via PI3K\nRalGDS\nNF-\u03baB", size = 2.5, color = "#1A5276", hjust = 0.5)

# Arrow from cross talk down to bottom box
p <- p + arr(rcx, rcx, 4.6, 4.35, blue_border)

# ===== BOTTOM CONVERGENCE BOX =====
# Both arrows converge to this
p <- p +
  arr(lcx, lcx, 2.6, 2.35, "#666666") +
  arr(rcx, rcx, 4.6, 4.35, blue_border)

# Redraw final arrow for left since we already drew above
# Bottom wide box
p <- p +
  box(1.2, 12.8, 1.55, 2.3, "#F0F0F0", "#666666") +
  lbl(7, 2.05, "Proliferation  \u00b7  Survival  \u00b7  Treatment resistance",
      size = 5, fontface = "bold", color = "#333333") +
  lbl(7, 1.76, "Same downstream phenotype — independent of direct RAS mutation",
      size = 3.3, color = "#555555")

# Arrows from column boxes to bottom box
p <- p +
  arr(lcx, lcx, 2.6, 2.3, "#666666") +
  arr(rcx, rcx, 4.6, 4.3, "#666666")

# ===== HYPOTHESIS BAR =====
p <- p +
  box(0.5, 13.5, 0.1, 1.35, navy_fill, navy_fill) +
  lbl(7, 0.88,
      "Hypothesis: ARHG mutations dysregulate Rho GTPase activity, feeding into RAS-adjacent signaling even without direct RAS mutations",
      size = 4, fontface = "bold.italic", color = "white") +
  lbl(7, 0.45,
      "ARHG mutations may be independent oncogenic events \u2014 not simply passengers to NRAS or KRAS",
      size = 3.3, color = "#BDC3C7")

# ===== SAVE =====
ggsave(
  file.path(BASE_DIR, "Presentations/arhg_ras_connection.png"),
  plot = p,
  width = W, height = H, dpi = 180, bg = "white"
)

cat("Saved arhg_ras_connection.png\n")
